File: apple/Sources/ReactNativeWatchOS/Modifiers/AccessibilityHidden.swift
import SwiftUI

/// SwiftUI `.accessibilityHidden(_:)`. When `true` (the default when the
/// param is absent), hides the view and its children from assistive
/// technologies.
enum RNWAccessibilityHiddenModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("accessibilityHidden") { view, params, _ in
            AnyView(view.accessibilityHidden(params.bool("value") ?? true))
        }
    }
}

File: src/modifiers/accessibilityHidden.ts
import { createModifier } from './createModifier';

/// SwiftUI `.accessibilityHidden(_:)`. When `true` (the default), hides the
/// view and its children from assistive technologies.
export function accessibilityHidden(value = true) {
  return createModifier('accessibilityHidden', { value });
}
